plugin/src/AlgoliaSync.php

```diff
@@ -385,10 +395,28 @@ private function elementValueFor(\Element $element, array $params, $full_related
             return $value;
         }
 
-        $value = null;
-        $this->zoo->event->dispatcher->notify($this->zoo->event->create($element, 'algolia:elementdata', ['value' => &$value]));
+        //$this->zoo->event->dispatcher->notify($this->zoo->event->create($element, 'algolia:elementdata', ['value' => &$value]));
+
+        return $this->notify('onAlgoliaElementData', [$element]);
+
+    }
+
+    private function notify($event_name, $data)
+    {
+        $dispatcher = \JEventDispatcher::getInstance();
```

Comment:
credo tu possa fare `$this->application->getDispatcher()`. `JEventDispatcher` è deprecato e viene rimosso nella v4 di Jooml